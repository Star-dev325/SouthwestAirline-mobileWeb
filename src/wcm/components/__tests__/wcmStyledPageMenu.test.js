import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import WcmStyledPageMenu from 'src/wcm/components/wcmStyledPageMenu';

describe('WcmStyledPageMenu component', () => {
  let onClickStub;

  beforeEach(() => {
    onClickStub = jest.fn();
  });

  it('should render an img component', () => {
    const { container } = createComponent();

    expect(container).toMatchSnapshot();
  });

  it('should trigger onClick callback after user click the component', () => {
    const { container } = createComponent();

    fireEvent.click(container.querySelector('a'));

    expect(onClickStub).toHaveBeenCalledWith({
      link_type: 'app',
      target: 'airbooking'
    });
  });

  const createComponent = () => render(
    <WcmStyledPageMenu
      altText="test"
      image="test.png"
      linkType="app"
      onClick={onClickStub}
      target="airbooking"
      type="menu"
    />
  );
});
