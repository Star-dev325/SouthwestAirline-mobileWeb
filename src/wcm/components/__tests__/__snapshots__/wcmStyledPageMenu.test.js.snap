// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WcmStyledPageMenu component should render an img component 1`] = `
<div>
  <a>
    <img
      alt="test"
      class="image fit mt1"
      src="test.png"
    />
  </a>
</div>
`;
